### PR TITLE
Release v0.8 docs

### PR DESCRIPTION
**Pull request by unityjeffrey**
Doc review.  

- Went through changes to existing docs this release and made updates.
- Reviewed new docs - custom protobuf.
- Added a Training Using Concurrent Unity Instances Page
- Minor tweaks here and there

### CODE REVIEW

File: docs/Migrating.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+
+### Important Changes
+* We have split the Python packges into two seperate packages `ml-agents` and `ml-agents-envs`
+
+#### Steps to Migrate
+* If you are installing via PyPI, there is no change.
+* If you intend to make modifications to `ml-agents` or `ml-agents-envs` please check the Installing for Development in the [Installation documentation](Installation.md).
+
+#### Steps to Migrate
+* Make sure to remove the `ENABLE_TENSORFLOW` flag in your Unity project settings
```
Comment (xiaomaogy) [inline docs/Migrating.md]: We don't need this in from v0.7 to v0.8.

**Comment by xiaomaogy** [comment]
Looks good to me aside from the comments. Remove that then merge.

File: docs/Training-ML-Agents.md [hunk 2]
```diff
@@ -134,12 +134,11 @@ environment, you can set the following command line options when invoking
   [Academy Properties](Learning-Environment-Design-Academy.md#academy-properties).
 * `--train` – Specifies whether to train model or only run in inference mode.
   When training, **always** use the `--train` option.
-* `--num-envs` - Specifies the number of parallel environments to collect
+* `--num-envs=<n>` - Specifies the number of concurrent, parallel Unity environment instances to collect
```
Comment (harperj) [inline docs/Training-ML-Agents.md]: Concurrent and parallel are saying the same thing here.

File: docs/Training-ML-Agents.md [hunk 3]
```diff
@@ -134,12 +134,11 @@ environment, you can set the following command line options when invoking
...
   experiences from when training. Defaults to 1.
-* `--base-port` - Specifies the starting port for environment workers. Each Unity
-  environment will use the port `(base_port + worker_id)`, where the worker ID
-  are sequential IDs given to each environment from 0 to `num_envs - 1`.
-  Defaults to 5005.
+* `--base-port` - Specifies the starting port for each concurrent, parallel Unity environment instance. Each instance will use the port `(base_port + worker_id)`, where the `worker_id`
```
Comment (harperj) [inline docs/Training-ML-Agents.md]: This is a bit confusing: "the starting port for each concurrent, parallel Unity environment instance".  This is the starting port from which all other ports are given sequential port numbers.  Not the starting port for each.

**Comment by harperj** [comment]
Couple of comments, but no show stoppers

**Comment by eshvk** [comment]
@unityjeffrey Can I squash and merge this?

**Pull request merged**